src/MSBuild.SDK.SystemWeb.CommonFiles/Sdk/MSBuild.SDK.SystemWeb.Common.DefaultPackages.targets

```diff
@@ -1,26 +1,37 @@
-﻿<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <!-- If we are using a central package management system such as Microsoft.Build.CentralPackageVersions do not include package versions -->
-  <PropertyGroup>
-    <!-- Requires Microsoft.Build.CentralPackageVersions Version 2.1.1 or higher
-         See https://github.com/microsoft/MSBuildSdks/tree/main/src/CentralPackageVersions  -->
-    <ApplySDKDefaultPackageVersions Condition="'$(UsingMicrosoftCentralPackageVersionsSdk)'=='true'">false</ApplySDKDefaultPackageVersions>
-    <!-- Turn of if using Nuget 6.2.0+ CPM 
-         See https://learn.microsoft.com/en-us/nuget/consume-packages/Central-Package-Management
-         Addresses #49 -->
-    <ApplySDKDefaultPackageVersions Condition="'$(ManagePackageVersionsCentrally)'=='true'">false</ApplySDKDefaultPackageVersions>
-    <ApplySDKDefaultPackageVersions Condition="'$(ApplySDKDefaultPackageVersions)'==''">true</ApplySDKDefaultPackageVersions>
-  </PropertyGroup>
+﻿<!-- 
+*************************************************************************************************************************
+MSBuild.SDK.SystemWeb.CommonFiles.DefaultPackages.targets
+    Ensure that the desirable PackageReference items are present if configured to do so 
+    * Do this After the "ProjectFile" so we can detect if the project has already explicitly included these packages
+    * Do this Before the "Directory.Build.targets" so CentralPackageVersions can act on these packages
+    * Which means before the entire import chain
+      Microsoft.NET.Sdk/Sdk/Sdk.props
+        -> Microsoft.CSharp.targets 
+          -> Microsoft.CSharp.CurrentVersion.targets 
+              -> Microsoft.Common.targets 
+                  -> Directory.Build.targets
+*************************************************************************************************************************
+-->
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
   <!-- Include compilers unless ExcludeSDKDefaultPackages property is set -->
-  <ItemGroup Condition="'$(ExcludeSDKDefaultPackages)'=='false'">
-    <PackageReference Include="Microsoft.Net.Compilers.Toolset" PrivateAssets="All" GeneratePathProperty="true"/>
-    <PackageReference Include="Microsoft.CodeDom.Providers.DotNetCompilerPlatform" />
-  </ItemGroup>
+  <ItemGroup Condition=" '$(ExcludeSDKDefaultPackages)'=='false' ">
+    <PackageReference Include="Microsoft.Net.Compilers.Toolset"
+                      Condition="'$(MicrosoftNetCompilersToolset_Version)'!=''"
```

Comment:
I think these shouldn't have the condition of the version on them, as they may not have versions set if using CPM/CPV.

Would it be better to create a new item group, e.g. `SDKDefaultPackageReference` with these, then at the end of common do a
```xml
<PackageReference Include="@(SDKDefaultPackageReference)" 
                  Exclude="@(PackageReference)" 
                  Condition="'$(ExcludeSDKDefaultPackages)'=='false'">
  <Version /><!-- Make sure we don't apply version's here -->
  <SystemWebSdkIncludedPackage>true</SystemWebSdkIncludedPackage>
  <SystemWebSdkIncludedPackageVersion>%(Version)</SystemWebSdkIncludedPackageVersion>
</PackageReference>
```
or is that adding complexity that it doesn't need? (Would need to import common at the end of razor/sdk defaultpackages instead of the start.)

I just think that having simple to manage lines for the default package refs (and any custom properties such as `PrivateAssets`, or `GeneratePathProperty`) would make it simpler to read and maintain - e.g. if adding a new reference.

```xml
<ItemGroup>
    <SDKDefaultPackageReference Include="Microsoft.Net.Compilers.Toolset"
                                Version="$(MicrosoftNetCompilersToolset_Version)"
                                PrivateAssets="All"
                                GeneratePathProperty="true" />
    <SDKDefaultPackageReference Include="Microsoft.CodeDom.Providers.DotNetCompilerPlatform"
                                Version="$(MicrosoftCodeDomProvidersDotNetCompilerPlatform_Version)" />
</ItemGroup>
```

Reply:
## Conditional check on named version property
That Conditional on the version property is a holdover from back before your recent updates.

Previously it looked like this
```Xml
  <ItemGroup Condition="'$(ExcludeASPNetCompilers)'=='false' AND '$(UsingMicrosoftCentralPackageVersionsSdk)'!='true'">
    <PackageReference Include="Microsoft.Net.Compilers.Toolset" Version="$(MicrosoftNetCompilersToolset_Version)" PrivateAssets="All" Condition="'$(MicrosoftNetCompilersToolset_Version)'!=''"/>
    <PackageReference Include="Microsoft.CodeDom.Providers.DotNetCompilerPlatform" Version="$(MicrosoftCodeDomProvidersDotNetCompilerPlatform_Version)" Condition="'$(MicrosoftCodeDomProvidersDotNetCompilerPlatform_Version)'!=''" />
  </ItemGroup>
```
I had assumed that it was an intentional way of allowing project owners to exclude an individual package (clear out property for single named package), so I had included that old conditional in this rendition.  But we can probably go with  `ExcludeSDKDefaultPackages` as the sole decision point.

## Separate ItemGroup
I had thought about having some itemgroup that is the list of packages we will work on, but we already have named properties for each package and that is a fairly straight forward way for project owners to tweak if they want to.

In general, I think whatever these end up being, they still needs to be "incorporated" in 2 phases though
1. Ensure that the items are added to `PackageReference` item group (without adding duplicates)... so after projectfile, but before Directory.Build.targets
2. Ensure appropriate Version metadata is in place... this could be `PackageReference` or `PackageVersion` depending on which mechanism is at work... so after Directory.Build.targets

I'll think on it some more